components/org.wso2.carbon.status.dashboard.core/src/main/java/org/wso2/carbon/status/dashboard/core/dbhandler/StatusDashboardMetricsDBHandler.java

```diff
@@ -608,7 +608,12 @@ public List selectWorkerAggregatedThroughput(String workerId, long timeInterval,
         try {
             stmt = conn.prepareStatement(query);
             for (int i = 1; i <= parameters.length; i++) {
-                stmt.setString(i, parameters[i - 1]);
+                //currently always the queries has two long values at the end and others are string values.
+                if(i <= parameters.length - 2){
```

Comment:
Shall we use an object array and then use instance of to achieve this. Yes it will add additional instance of check but will eliminate Long.toString and Long.parseLong invocations. WDYT?

Reply:
+1 will do